luigi/contrib/kubernetes.py

```diff
@@ -157,18 +157,35 @@ def max_retrials(self):
         """
         return self.kubernetes_config.max_retrials
 
+    @property
+    def delete_on_success(self):
+        """
+        Delete the Kubernetes workload if the job has ended successfully.
+        """
+        return True
```

Comment:
This should be False IMO. In many cases it is important to keep track of what was ran on the infrastructure along with the logs. If the Pod is deleted, then there is no way to do auditing. I am open for discussion though.

Reply:
I don't believe that's how Kubernetes Jobs are supposed to be used - logs should be used for this purpose. Additionally, when Luigi DAGs are recurrently scheduled, not removing the successful jobs creates clutter and forces an extra cleanup step. Also, please note that if a job is not successful, the pod will not be removed - maintaining troubleshooting capabilities when needed.

Reply:
Yep, you are totally right!